driver-core/src/main/com/mongodb/client/gridfs/model/GridFSFileCodecProvider.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2016 MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.client.gridfs.model;
+
+import org.bson.codecs.Codec;
+import org.bson.codecs.configuration.CodecProvider;
+import org.bson.codecs.configuration.CodecRegistry;
+
+/**
+ * A provider of codecs for {@link GridFSFile}s.
+ *
+ * @since 3.3
+ */
+public final class GridFSFileCodecProvider implements CodecProvider {
```

Comment:
More generally useful would be something like:

public final class CodecProviders {
    <T> public static CodecProvider fromCodec(Codec<T> codec);
}

that returns a private implementation that is a provider for a single Codec (can get the Class instance from the getEncoderClass method)